setup.py

```diff
@@ -121,4 +119,14 @@
     scripts=scripts,
     data_files=data_files,
     entry_points=entry_points,
+    python_requires=">=3.10,<3.14",
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.10",
```

Comment:
@xylar @lee1043 @zhang40 We've decided to keep support for Python 3.10 in the next E3SM-Unified? Just making sure as I have not removed 3.10 support in other E3SM tools yet.

Reply:
I completely agree with you. We should ensure that we're up-to-date with the rest of the scientific Python ecosystem, rather than lag behind and implement temporary (and often difficult/hacky) workarounds.

If we make the decision to drop Python 3.10, I will follow up shortly with PRs for other E3SM packages.